src/twisted/conch/ssh/keys.py

```diff
@@ -1241,8 +1396,20 @@ def verify(self, signature, data):
                 k = k.public_key()
             verifier = k.verifier(
                 signature, hashes.SHA1())
-        else:
-            raise BadKeyError("unknown key type %s" % (self.type(),))
+
+        elif keyType == 'EC':
```

Comment:
What should *really* be happening here - although this refactoring should happen in a follow-up branch, we need not block on it - is that this code should be doing `self._keyImpl.verify(...)`, and then we should have 3 different classes for `_keyImpl`, which wrap each of the different key types.  A class with this many if/elif trees is a code smell.

In the meanwhile, to compromise between coverage and quality, a [`# pragma: no branch` comment on the last `elif`](http://coverage.readthedocs.io/en/coverage-4.2/branch.html#structurally-partial-branches) ought to resolve the conflict between coverage and clarity.  Please file a ticket to refactor the guts of the if/elif trees out into their own separate classes though, and link to it from a subsequent comment, so we don't lose track of this refactoring.